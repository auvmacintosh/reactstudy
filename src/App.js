import React from 'react';
import 'normalize.css';

let log = console.log.bind(console);

class Item extends React.Component {

    get name() {
        return this.constructor.name + this.props.name;
    }

    componentWillReceiveProps() {
        log(this.name + ' will receive props')
    }

    componentWillUpdate() {
        log(this.name + ' will update')
    }

    componentWillUnmount() {
        log(this.name + ' will unmount');
    }

    componentWillMount() {
        log(this.name + ' will mount');
    }

    componentDidMount() {
        log(this.name + ' did mount')
    }

    shouldComponentUpdate(np, ns) {
        return this.props.name !== np.name;
    }

    render() {
        log(this.name + ' render')
        return <div>{this.props.name}</div>
    }
}

class App extends React.Component {
    constructor(props) {
        super(props);
        this.state = {xs: [1, 2, 3, 4, 5]}
    }

    handler = () => {
        this.setState({xs: [5, 1, 2, 3, 4]})
    }

    render() {
        return (
            <>
                {this.state.xs.map(x => <Item key={x} name={x}/>)}
                <button onClick={this.handler}>Click</button>
            </>
        )
    }
}

export default App;