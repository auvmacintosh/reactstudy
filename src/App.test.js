import React from 'react';
import ReactDOM from 'react-dom';
// import './setupTests';  // setupTests.js在test之前会被自动执行
import App from './App';
import renderer from 'react-test-renderer';
import {render} from 'react-testing-library';

// smoke test with only jest
test('renders without crashing', ()=> {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
})

// snapshot test with react-test-renderer
// put x prefix to skip the test
xtest("App changes the class when hovered", ()=> {
    const component = renderer.create(
        <App page="einfobank.net">hello</App>
    )
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

// unit test with react-testing-library
test('renders welcome message', () => {
    const { getByText } = render(<App>hello</App>);
    expect(getByText('hello')).toBeInTheDocument();
});
