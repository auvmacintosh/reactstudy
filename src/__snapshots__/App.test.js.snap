// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App changes the class when hovered 1`] = `
Array [
  "hello",
  <a
    className="leave"
    href="einfobank.net"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    leave
  </a>,
  <button
    onClick={[Function]}
  >
    wangboC1
  </button>,
]
`;
